native/src/components/MapView.tsx

```diff
@@ -12,49 +14,84 @@ const MapContainer = styled.View`
 const StyledMap = styled(MapboxGL.MapView)`
   width: 700px;
   height: 500px;
+  z-index: 1;
 `
 
 type MapViewPropsType = {
   boundingBox: BBox
   featureCollection: FeatureCollection
+  currentFeature?: Feature<Point>
 }
 
 const textOffsetY = 1.25
-const layerProps: SymbolLayerProps = {
-  id: 'point',
-  style: {
-    symbolPlacement: 'point',
-    iconAllowOverlap: true,
-    iconIgnorePlacement: true,
-    iconImage: ['get', 'symbol'],
-    textField: ['get', 'title'],
-    textFont: ['Roboto Regular'],
-    textOffset: [0, textOffsetY],
-    textAnchor: 'top',
-    textSize: 12
-  }
-}
+const layerId = 'point'
 
 // Has to be set even if we use map libre
 MapboxGL.setAccessToken(mapConfig.accessToken)
-const MapView = ({ boundingBox, featureCollection }: MapViewPropsType): ReactElement => {
+const MapView = ({ boundingBox, featureCollection, currentFeature }: MapViewPropsType): ReactElement => {
+  const mapRef = React.useRef<MapboxGL.MapView | null>(null)
+  const cameraRef = React.useRef<MapboxGL.Camera | null>(null)
+  const [activeFeature, setActiveFeature] = useState<Feature<Point> | null>(currentFeature ?? null)
+  const [showPopup, setShowPopup] = useState<boolean>(!!currentFeature)
+  const layerProps: SymbolLayerProps = {
+    id: layerId,
+    style: {
+      symbolPlacement: 'point',
+      iconAllowOverlap: true,
+      iconIgnorePlacement: true,
+      iconImage: ['get', 'symbol'],
+      textField: ['case', ['==', ['get', 'id'], activeFeature?.properties?.id ?? -1], ['get', 'title'], ''],
+      textFont: ['Roboto Regular'],
+      textOffset: [0, textOffsetY],
+      textAnchor: 'top',
+      textSize: 12
+    }
+  }
+
   const bounds = {
     ne: [boundingBox[2], boundingBox[3]],
     sw: [boundingBox[0], boundingBox[1]]
   }
 
+  // if there is a current feature use the coordinates if not use bounding box
   const defaultSettings: CameraSettings = {
-    zoomLevel: defaultViewportConfig.zoom,
-    bounds
+    zoomLevel: activeFeature?.geometry.coordinates ? detailZoom : defaultViewportConfig.zoom,
+    centerCoordinate: activeFeature?.geometry.coordinates,
+    bounds: activeFeature?.geometry.coordinates ? undefined : bounds
   }
+
+  const onPress = useCallback(
+    async (e: Feature) => {
+      const featureCollection: FeatureCollection<Point> = (await mapRef?.current?.queryRenderedFeaturesAtPoint(
+        [e.properties?.screenPointX, e.properties?.screenPointY],
+        undefined,
+        [layerId]
+      )) as FeatureCollection<Point>
+      if (featureCollection?.features?.length) {
+        const { features } = featureCollection
+        setActiveFeature(features[0])
+        setShowPopup(true)
+        const {
+          geometry: { coordinates }
+        } = features[0]
+        cameraRef?.current?.flyTo([coordinates[0], coordinates[1]])
+      } else {
+        setActiveFeature(null)
+        setShowPopup(false)
+      }
+    },
+    [cameraRef]
+  )
```

Comment:
```suggestion
  const onPress = useCallback(async (e: Feature<Geometry, GeoJsonProperties>) => {
    if (!mapRef?.current || !cameraRef?.current || !e.properties) {
      return
    }
    const featureCollection = await mapRef.current.queryRenderedFeaturesAtPoint(
      [e.properties.screenPointX, e.properties?.screenPointY],
      undefined,
      [layerId]
    )
    const feature = featureCollection?.features?.find((it): it is Feature<Point> => it.geometry.type === 'Point')

    if (feature) {
      setActiveFeature(feature)
      const {
        geometry: { coordinates }
      } = feature
      cameraRef.current.flyTo(coordinates)
    } else {
      setActiveFeature(null)
    }
  }, []
```
I think we should avoid typecasts and nullish coalesence operator as much as possible.

Reply:
Or is there any reason why you just included `cameraRef` in dependencies?

Reply:
cameraRef was suggested, i wouldn't have added it

Reply:
Nullish coalescing operator is "??" or am i wrong?
What you mean is "Optional Chaining" that means checking for null | undefined of objects which became a ECMA standard. I see in your suggestion the advantage no question, but there are lots of cases where it is the recommend way to go

Reply:
Yeah right, meant optional chaining, sorry :D